.github/workflows/preview-env-deploy.yml

```diff
@@ -0,0 +1,111 @@
+name: preview-env-deploy
+on:
+  pull_request:
+    types: [labeled, synchronize]
+
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+
+jobs:
+  deploy-preview:
+    if: github.event.pull_request.state != 'closed' && (contains( github.event.label.name, 'deploy') || contains( github.event.pull_request.labels.*.name, 'deploy'))
+    runs-on: ubuntu-22.04
+    timeout-minutes: 30
```

Comment:
no changes requested, just a note for @akeller -- this shouldn't be a problem as we're well under this with build times right now, but if times start to creep back up again, we might notice this.